06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,91 @@
+
+/**
+ * InvertedIndex class with constructor
+ * @class
+ */
+class InvertedIndex {
+
+  /**
+   * Creates an instance of InvertedIndex.
+   * @memberOf InvertedIndex
+   */
+  constructor() {
+    this.indexedFiles = {};
+    this.searchResults = {};
+  }
+
+  /** creates index and update the indexed files
+   * @param {object} uploadedFiles - object containing uploaded books
+   * @param {Sting} file - file name
+   * @return  {object}  - this.indexedFiles
+   */
+  updateIndexedFilesRecords(uploadedFiles, file) {
```

Comment:
@q-ode  i have refactored the codes here and the functions are now easy to understand